### PR TITLE
feat(#2125): add OyFilesystemSources class

### LINKED ISSUE #2125: SnippetTest.java:293-296: Introduce OyFilesystem instead...
The puzzle `1804-0c2079fd` from #1804 has to be resolved: 

https://github.com/objectionary/eo/blob/03bfca5169b891538c3e758e7b8d7506c5f81e48/eo-maven-plugin/src/test/java/org/eolang/maven/SnippetTest.java#L293-L296

The puzzle was created by rultor on 05-Jun-23. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
Add OyFilesystemSources class for testing purposes.

Closes #2125

---

## PR-Codex overview
This PR replaces `OyFake` with `OyFilesystem` in `SnippetTest.java` to simplify the code and allow for testing purposes. 

### Detailed summary
- Replaces `OyFake` with `OyFilesystem` in `SnippetTest.java`
- `OyFilesystem` is a new class that downloads object sources directly from the filesystem, useful for testing purposes
- `OyFake` is no longer used and can be removed

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
@mximp Could you have a look, please?

File: eo-maven-plugin/src/main/java/org/eolang/maven/objectionary/OyFilesystemSources.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+ * Objectionary stored locally in the filesystem.
+ * The aim of this class is to download object sources directly from the filesystem, for example,
+ * from the project itself. It is useful for testing purposes.
+ *
+ * @since 0.30
+ * @todo #2125:30min Add unit tests for OyFilesystemSources. The tests should check that
+ *  OyFilesystemSources can get an object from the filesystem and that it can check if an object
+ *  exists in the filesystem.
+ */
+public final class OyFilesystemSources implements Objectionary {
```
Comment (mximp) [inline eo-maven-plugin/src/main/java/org/eolang/maven/objectionary/OyFilesystemSources.java:43]: @volodya-lombrozo Does it really imply multiple sources per instance? Maybe `OyFilesystem` is better?

**Review by mximp — CHANGES_REQUESTED** [review]
@volodya-lombrozo I strongly believe that implemented functionality must be covered with tests. Otherwise (even with puzzle) we decrease quality of code within master branch.

**Comment by volodya-lombrozo** [comment]
@mximp As for tests - They will be implemented in the next puzzle. We can do it because the OyFilesystem is already being used in other tests, which proves the correct functioning of the class. Moreover, if you check the code coverage, you will see that it remains unchanged. Thus, we can conclude that the quality level remains the same. Tests here are required more for understanding the behavior of the class. It's just an enhancement.

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you merge that changes, please?

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**